### PR TITLE
Fix ENABLE_DBM_NEW, and build it in the CI

### PR DESCRIPTION
**Pull request by mikucionisaau**
Test attempting to expose memory issues with fed allocation.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -56,7 +57,7 @@ jobs:
           export CTEST_OUTPUT_ON_FAILURE=1
           mkdir build
           cd build
-          cmake -DTESTING=yes -DSTATIC=$STATIC  -DASAN=${{ matrix.asan }} -DUBSAN=${{ matrix.ubsan }} -DCMAKE_BUILD_TYPE=Release ..
+          cmake -DTESTING=yes -DSTATIC=$STATIC  -DASAN=${{ matrix.asan }} -DUBSAN=${{ matrix.ubsan }} -DCMAKE_BUILD_TYPE=Release -DENABLE_DBM_NEW=${{ matrix.newdbm }} ..
```
Comment (mikucionisaau) [inline .github/workflows/build.yml:60]: maybe we should have just one flag for all the sanitizers: either all on or all off, it makes no sense to test them separately.

**Review by RasmusRendal — APPROVED** [review]

**Comment by mikucionisaau** [comment]
I think `DBMAllocator` should go, I had lot's of issues with it before and I do not know the purpose of its existence: 
it could have been a performance experiment (providing caching/recycling of released DBMs etc), or compatibility with C (what's `malloc`ed has to be `free`d and what's `new`ed have to be `delete`d).
The allocator performance is no longer an issue (it's fast), so we should not invent yet another garbage maintenance layer. Then the de/allocations should be self-contained within the library (should part of implementation and should not disturb the header, moreover it seems that it is C++ only code anyway).

**Comment by mikucionisaau** [comment]
@yrke welp, how can we merge this?
It seems that some workflows were renamed, the new ones are stuck with "Waiting for status to be reported", but merge requires them.
Is there a button I can press? I do not see `Settings` tab, so maybe I am lacking some permissions?

**Comment by yrke** [comment]
@mikucionisaau I disabled the CI requirement to allow merge

**Pull request merged**